File: backend/src/main/java/com/NBE_4_5_2/Team5/domain/notification/entity/Notification.java
package com.NBE_4_5_2.Team5.domain.notification.entity;

import java.util.UUID;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Notification {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	private String userId;
	private boolean global;
	private String content;

	public Long getId() {
		return id;
	}

	public String getUserId() {
		return userId;
	}

	public boolean isGlobal() {
		return global;
	}

	public String getContent() {
		return content;
	}

	public Notification() {
	}

	public Notification(String userId, boolean global, String content) {
		this.userId = userId;
		this.global = global;
		this.content = content;
	}
}

File: backend/src/main/java/com/NBE_4_5_2/Team5/domain/notification/entity/Notification.kt
package com.NBE_4_5_2.Team5.domain.notification.entity

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.persistence.*
import java.util.logging.Logger.global

@Entity
class Notification(
    @Column(name="user_id")
    @JsonProperty(value = "userId")
    private var _userId: String?,
    @Column(name = "global")
    @JsonProperty(value="global")
    private var _isGlobal: Boolean = false,
    @Column(name = "content")
    @JsonProperty(value="content")
    private var _content: String?=null
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    @JsonProperty(value = "id")
    private var _id: Long? = null
    val id: Long
        get() = _id!!

    val userId: String?
        get()=_userId

    val global: Boolean
        get()=_isGlobal

    val content: String?
        get()=_content


    constructor(global: Boolean, content: String) : this(
        null,
        _isGlobal = global,
        _content = content
    )
    constructor(id:Long, userId:String?, global:Boolean, content:String?):this(
        _userId=userId,
        _isGlobal=global,
        _content= content,
    ){
        _id=id
    }
}
